client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -106,41 +106,55 @@ tie.factory('PythonCodeRunnerService', [
       });
     };
 
-    var _sendCodeToServer = function(code, urlFragment) {
+    var _compileCodeAsync = function(code) {
       clearOutput();
-      return $http({
-        url: SERVER_URL + '/ajax/' + urlFragment,
-        method: 'POST',
-        data: {
-          code: code,
-          language: 'python'
+      var data = {
+        code: code,
+        language: 'python'
+      };
+      return $http.post(SERVER_URL + '/ajax/compile_code', data).then(
+        function(response) {
+          _processCodeCompilationServerResponse(response, code);
         }
-      }).then(function(response) {
-        _processServerResponse(response, code, urlFragment);
-      });
+      );
+    };
+
+    var _processCodeCompilationServerResponse = function(response, code) {
```

Comment:
Pass in responseData instead, so that this function doesn't have access to unnecessary stuff. Ditto for _processCodeExecutionServerResponse.